core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -90,10 +90,19 @@ private static Map<String, String> unreservedProperties(Map<String, String> rawP
   private static Map<String, String> persistedProperties(Map<String, String> rawProperties) {
     Map<String, String> persistedProperties = Maps.newHashMap();
 
+    String format =
+        rawProperties.getOrDefault(
```

Comment:
If file format is not set in `rawProperties` but at write time, parquet compression codec will still be persisted since the default format is `parquet`

Reply:
Not sure if I follow, isn't that the right behavior? If the default format is parquet then the parquet compression codec should be set.

Reply:
The format can be set to ORC at write time. The parquet compression codec is already persisted then.

Reply:
Ah I see, you're talking about the dynamic write configuration options that exist at the engine level ("write-format") for Spark and Flink. Yeah that's true, this doesn't really solve that case which would still be confusing for users. I'll think a bit more about a good solution for that then.

Reply:
Well if I think about it, what we are persisting at the table property level is a default for the table. Whether or not the Flink/Spark write options override at runtime doesn't really matter. For example, if they override with ORC in the runtime option and we end up persisting parquet compression doesn't really matter because if they later perform a write *without* the runtime operation we would be writing ZSTD parquet because we would be falling back to the table property.

I could see it being confusing though if everyone relied on the engine dynamic config for all their writes (always specifying in the write option) rather than the table property.

Reply:
Then I'm wondering if this can fix #9490. An alternative is to filter out parquet compression if the write format is not parquet when users are checking properties.

Reply:
I believe it should since in Trino the only way to specify the format is just at the table property.
Trino does not have any session properties in the Iceberg connector for overriding the file format at runtime as far as I can tell.  https://github.com/trinodb/trino/blob/master/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSessionProperties.java#L57 

I'll run an end to end test with Trino and this PR to make sure this PR does solve the problem in #9490.

Reply:
![Screenshot from 2024-01-19 09-22-10](https://github.com/apache/iceberg/assets/87500546/32641a91-58c7-4548-a2eb-24387af51c6e)
![Screenshot from 2024-01-19 10-25-19](https://github.com/apache/iceberg/assets/87500546/dc135124-4546-489b-aedd-20450c13e95a)

Here is a smoke test with ORC format in Trino before and after the change. Before the change the test passes because Parquet compression was defined with ZSTD, after there's no such table property.